torch/distributed/_composable/fsdp/_fsdp_common.py

```diff
@@ -53,6 +53,22 @@ def __post_init__(self):
         super(DDPMeshInfo, self).__post_init__()
 
 
+@dataclass
+class ParamModuleInfo:
```

Comment:
Nit: Is ``ParamModuleInfo`` used in used in ``FSDPParam`` only ? wonder if we should put it in ``_fsdp_common.py`` or ``_flat_param.py``

Reply:
This is a good point! Since `ParamModuleInfo` is used in both `FSDPParam` and `FSDPParamGroup`, I had `ParamModuleInfo` in `_fsdp_common.py`. However, I realize now that this is not entirely consistent (e.g. `ShardedState` is used in both but is in `_fsdp_param.py`), and this inconsistency is a consequence of how the code evolved over time.

Do you prefer these to be in `_fsdp_param.py`?

If so, I might prefer to move it in a separate PR because changing the imports causes so many rebase conflicts 😢 , but I can do it here if desired too.

Reply:
separate PR works for me. other files import from `_fsdp_param.py`since it ultimately serves ``FSDPParam``

Reply:
To be good, I am going to try to fix this directly here and handle the rebase conflicts 😅